llvm/lib/SYCLLowerIR/SYCLJointMatrixTransform.cpp

```diff
@@ -21,6 +20,9 @@ namespace {
 
 static constexpr char ACCESS_CHAIN[] = "_Z19__spirv_AccessChain";
 static constexpr char MATRIX_TYPE[] = "spirv.CooperativeMatrixKHR";
+static constexpr char MATRIX_LAYOUT[] =
+    "_ZN4sycl3_V16detail26joint_matrix_layout_to_spvENS0_"
+    "3ext6oneapi12experimental6matrix6layoutE";
```

Comment:
if something is changed in SYCL (e.g. namespace version V1->V2, layout is moved out of experimental namespace), then this match will not work anymore. Is it possible to make it less fragile (or add some test to catch that)?

Reply:
> or add some test to catch that

I'd say we already have the tests - any test in test-e2e/Matrix directory, just need to start compiling them with O0.

> if something is changed in SYCL (e.g. namespace version V1->V2, layout is moved out of experimental namespace)

I'm not keen to demangle every function in the module and compare it with `joint_matrix_layout_to_spv`. Also using StringRef's `contains` API seems to be fragile for me. Alternative approach would be to make `joint_matrix_layout_to_spv` to be `extern "C"` in SYCL headers and check it's name without mangling here. It's not "C++" way, but we don't expect any overloads for the function, so it might be fine. WDYT?

Reply:
I think `extern "C"` approach should be OK together with some comment, why `extern "C"` is used. So that if anyone in the future wants to update this function (its name or remove `extern "C"`), they know the side effects.